src/main/java/com/exactpro/epfast/template/assertion/InstrEqualityChecker.java

```diff
@@ -0,0 +1,160 @@
+package com.exactpro.epfast.template.assertion;
+
+import com.exactpro.epfast.template.*;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.exactpro.epfast.template.assertion.OpEqualityChecker.areSameOperators;
+import static com.exactpro.epfast.template.assertion.TemplateAssert.areSameIdentities;
+import static com.exactpro.epfast.template.assertion.TemplateAssert.areSameReferences;
+
+public class InstrEqualityChecker {
+
+    static boolean areSameInstructions(List<? extends Instruction> actual, List<? extends Instruction> instr) {
+        if (actual == instr) {
+            return true;
+        }
+        if (actual.size() != instr.size()) {
+            return false;
+        }
+
+        for (int i = 0; i < actual.size(); i++) {
+
+            Instruction actualItem = actual.get(i);
+            Instruction instrItem = instr.get(i);
+            if (actualItem != instrItem) {
+                if (actualItem.getClass() != instrItem.getClass()) {
+                    return false;
+                }
+
+                switch (actualItem.getClass().getSimpleName()) {
```

Comment:
I see your desire to use `Map` and create more readable code..
In no-map case `instanceof` should be used instead of `switch`:
```java
if (actualItem instanceof TemplateRef) {
}
else if (actualItem instanceof TemplateRef) {
}
```
It can be done with `Map` but there are some complications. If you want I can tell you more when coronovirus situation resolves and we're back in the office.